modules/ROOT/pages/storage.adoc

```diff
@@ -174,3 +174,63 @@ storage:
+
+== Disk Layout
+
+All Fedora CoreOS systems start with the same disk image which varies slightly between architectures based on what is needed for bootloading. On first boot the root filesystem is expanded to fill the rest of the disk. The disk image can be customized using Fedora CoreOS configs to repartition the disk and create/reformat filesystems. Bare metal installations are not different; the installer only copies the raw image to the target disk and injects the specified config into `/boot` for use on first boot.
+
+NOTE: The boot partition cannot be reformatted, deleted, or moved.
+
+NOTE: See xref:#_reconfiguring_the_root_filesystem[Reconfiguring the root filesystem] for examples regarding the supported changes to the root partition.
+
+=== Partition Tables
+
+Using partition numbers to refer to specific partitions is discouraged and labels or UUIDs should be used instead. Fedora CoreOS reserves the `boot`, `root`, `BIOS-BOOT` and `EFI-SYSTEM` labels. Creating partitions or filesystems with those labels is not supported.
+
+=== x86_64 Partition Table
+
+The x86_64 disk image is GPT formatted with a protective MBR. It supports booting via both BIOS and UEFI (including secure boot).
+
+.Partition Table for x86_64
+|============================================================================================
+| Number | Label      | Description                                          | Partition Type
+| 1      | boot       | Contains GRUB configuration, kernel/initramfs images | ext4
+| 2      | EFI-SYSTEM | Contains EFI GRUB image and secure boot shim         | FAT32
+| 3      | BIOS-BOOT  | Contains BIOS GRUB image                             | raw data
+| 4      | root       | Contains the root filesystem                         | xfs
+|============================================================================================
+
+The EFI-SYSTEM partition can be deleted or reformatted when BIOS booting. Similarly, the BIOS-BOOT partition can be deleted or reformatted when
+EFI booting.
+
+== Mounted Filesystems
+
+Fedora CoreOS uses OSTree, which contains multiple deployments of the same OS. Each deployment contains a full root filesystem which gets mounted to / on boot. All deployments of the same OS share the same /var, which is kept separately.
+
+=== State in /var
+
+OSTree requires all state be kept in `/var`. Traditional places that might hold state (e.g. `/home`, or `/srv`) are symlinks to directories in `/var` (e.g. `/var/home` or `/var/srv`). Creating new directories under the root directory is not supported and disallowed by the immutable attribute. The contents of `/var` are located at `/ostree/deploy/fedora-coreos/deploy/var` by default, though moving `/var` to it's own filesystem is also supported using Fedora CoreOS configs.
+
+=== Read only /usr
+
+OSTree mounts `/usr` read only. Making changes to `/usr` or remounting it read/write is not supported.
```

Comment:
I'd say this more strongly - the only places admins should be changing are in `/etc` and `/var` and maybe in very rare cases `/boot`.

(Notably `/` has the immutable bit, this relates to https://github.com/coreos/fedora-coreos-tracker/issues/270 )